### ISSUE TITLE
Reduction method in RunUMAP

### ISSUE THREAD

**Issue by rgdani** [satijalab/seurat#9160]
Hi! When I put reduction = "umap" why does it say subscript out of bound, but when I did not specify the reduction method, the verbose says that it uses method 'umap'. Here, I tried specifying the RunUMAP reduction method to be 'umap' (below is the code)
```
endo <- subset(merged_seurat_filtered, idents = c("Epithelium A", "Epithelium B", "Endothelium"))
View(endo)
DimPlot(endo)
endo <- ScaleData(endo, verbose = TRUE)
endo <- RunPCA(endo, npcs = 30, verbose = TRUE)
ElbowPlot(endo)
endo <- FindNeighbors(endo, reduction = "pca", dims = 1:15)
endo <- FindClusters(endo, resolution = 0.2)
endo <- RunUMAP(object = endo, dims = 1:15)
DimPlot(endo, label = TRUE)
```
And here is the output: 
![Screenshot 2024-07-30 at 14 28 19](https://github.com/user-attachments/assets/732cb6fe-749d-49eb-9c97-aec9fa6f68a6)

But when I did not specify the reduction method in the RunUMAP, it says that it uses method 'umap'
![Screenshot 2024-07-30 at 14 29 04](https://github.com/user-attachments/assets/4bf2872a-e300-4c68-9ced-3b2a795c0dcd)

Any explanation is well received! Thank you!

**Comment by Alexis-Varin** [comment]
Hi, you are misinterpreting the `reduction` parameter of `RunUMAP()`, this isn't the same as the `reduction` parameter from `DimPlot()` for example which has the same name but which pulls the embeddings from that `DimReduc`.

Here `reduction` in `RunUMAP()` refers to which `DimReduc` already present in the object will be used to calculate UMAP from. If you put 'umap', you are saying to the function to calculate UMAP from 'umap', and you get subscript out of bounds not because it is not there, apparently it is in your object, but because you are telling the function to pull 15 dimensions from 'umap' when it only contains 2 (there is always only 2 UMAP dimensions).

When you don't set the `reduction` parameter in `RunUMAP()`, it will pull the embeddings from 'pca' by default, which will work.

So in conclusion, `reduction` in `RunUMAP()` needs to be either not set and therefore kept to 'pca' (if you didn't integrate) or set to the `new.reduction` parameter in `IntegrateLayers()` (for example, 'integrated.cca' in [the tutorial](https://satijalab.org/seurat/articles/integration_introduction#perform-integration).
